File: array/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;
        int longest=0;
        unordered_set<int> st(nums.begin(),nums.end());
        for(int num:st){
            if(st.find(num-1)==st.end()){
                int length=1;
                int next=num+1;
                while(st.find(next)!=st.end()){
                    length++;
                    next++;
                }
                longest=max(length,longest);
            }
        }
        return longest;

    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int i : nums) {
            s.add(i);
        }

        int longest = 0;

        for (int i : s) {
            if (!s.contains(i - 1)) {
                int curr = i;
                int length = 1;

                while (s.contains(curr + 1)) {
                    curr++;
                    length++;
                }

                longest = Math.max(longest, length);
            }
        }
        return longest;
    }
}